### PR TITLE
feat: implement current developer system as a separate mcp server

### PR DESCRIPTION
**Pull request by salman1993**
this is still not complete. the tools are working. resource is not working because i think its currently implemented as a static resource but we might need a resource template (dynamic) - not sure yet

- [x] fix resources
- [x] unit tests
- add other tools - list_windows, screen_capture (separate PR: https://github.com/block/goose/pull/527/files)

tested with MCP inspector:
```
npx @modelcontextprotocol/inspector cargo run -p developer
```

✅  tools 
Screenshot 2024-12-20 at 4 31 54 PM

Screenshot 2024-12-20 at 4 31 31 PM

✅  resources
Screenshot 2024-12-23 at 12 26 52 PM

### CODE REVIEW

**Comment by michaelneale** [comment]
this is very cool - and does make me think - do we want a way to run in-process systems for things are are pretty much always bundled? (but still be a system which could be used out of process)

File: crates/developer/src/errors.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
+use mcp_core::handler::{ResourceError, ToolError};
```
Comment (baxen) [inline crates/developer/src/errors.rs]: seems like file this should maybe be part of more central library? but i guess reasonable to wait until we have another (although @michaelneale has a new system that would need these i think)

File: crates/developer/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,1086 @@
...
+        let result = self.bash(args).await; // adapt your logic from DeveloperSystem
+        self.map_agent_result_to_value(result)
+    }
+
+    async fn call_text_editor(&self, args: Value) -> Result<Value, ToolError> {
+        let result = self.text_editor(args).await; // adapt from DeveloperSystem
+        self.map_agent_result_to_value(result)
+    }
+
+    // Convert AgentResult<Vec<Content>> to Result<Value, ToolError>
```
Comment (baxen) [inline crates/developer/src/main.rs]: nit: i haven't thought about it too hard, but might be better to just rework the old code a bit to create ToolError?

File: crates/developer/src/errors.rs (hunk 1 above)
Comment (baxen) [reply crates/developer/src/errors.rs]: Actually after reading, maybe we remove this entirely and only use ToolError from mcp_core

**Review by baxen — APPROVED** [review]
LGTM

**Pull request merged**